eosdactoken/eosdactoken.cpp

```diff
@@ -153,18 +155,44 @@ void eosdactoken::burn(account_name from, asset quantity ) {
     });
 }
 
-void eosdactoken::memberreg(name sender, string agreedterms) {
+  void eosdactoken::newmemterms(string ipfs) {
+    // member terms is expected to be an sha2656 IPFS
+    // print("In newmemterms(). ");
+
+    require_auth( _self );
+
+    // sample IPFS: QmXjkFQjnD8i8ntmwehoAHBfJEApETx8ebScyVzAHqgjpD
+    eosio_assert( !ipfs.empty(), "Member terms cannot be empty." );
+    eosio_assert( ipfs.length() == 46 , "Member terms sha256 IPFS is expected to be 46 character long." );
+    eosio_assert( ipfs[0] == 'Q' && ipfs[1] == 'm' , "Member terms sha256 IPFS is expected to begin with 'Qm'." );
+
+    // guard against duplicate of latest
+    if (memberterms.begin() != memberterms.end()) {
+      eosio_assert( ipfs != (--memberterms.end())->ipfs, "Next member terms cannot be duplicate of the latest." );
+    }
+
+    uint64_t next_version = (memberterms.begin() == memberterms.end() ? 0 : (--memberterms.end())->version) + 1;
+
+    memberterms.emplace(_self, [&](terms& term) {
+        term.ipfs = ipfs;
+        term.version = next_version;
+      });
+  }
+
+void eosdactoken::memberreg(name sender) {
```

Comment:
If we take this approach we are not proactively getting the sender to agree to a particular IPFS hash terms but rather the sender has agreed to whatever the current terms are when this is processed. This means the UI could potentially be out of date with the chain and the user would never know they have agreed to the updated terms. If instead they sent an IPFS hash as part of the transaction then the code could check for equality with the latest terms and reject if not equal. Still I really like the space saving using the indexed table for storage. If this issue has been flogged to death please disregard this comment. Otherwise the code looks very nice and I would be happy to merge.

Reply:
Very good point and agree with your assessment. Am making the change now.